contracts/YieldLeverBase.sol

```diff
@@ -44,13 +72,358 @@ contract YieldLeverBase is IERC3156FlashBorrower {
         giver = giver_;
     }
 
+    /// @notice Approve maximally for an fyToken.
+    /// @param seriesId The id of the pool to approve to.
+    function approveFyToken(bytes6 seriesId) external {
+        IPool(LADLE.pools(seriesId)).fyToken().approve(
+            address(LADLE),
+            type(uint256).max
+        );
+    }
+
+    /// @notice Invest by creating a levered vault. The basic structure is
+    ///     always the same. We borrow FyToken for the series and convert it to
+    ///     the yield-bearing token that is used as collateral.
+    /// @param ilkId The ilkId to invest in. This is often a yield-bearing
+    ///     token, for example 0x303400000000 (WStEth).
+    /// @param seriesId The series to invest in. This series doesn't usually
+    ///     have the ilkId as base, but the asset the yield bearing token is
+    ///     based on. For example: 0x303030370000 (WEth) instead of WStEth.
+    /// @param amountToInvest The amount of the base to invest. This is denoted
+    ///     in terms of the base asset: USDC, DAI, etc.
+    /// @param borrowAmount The amount to borrow. This is denoted in terms of
+    ///     debt at maturity (and will thus be less before maturity).
+    /// @param minCollateral Used for countering slippage. This is the minimum
+    ///     amount of collateral that should be locked. The debt is always
+    ///     equal to the borrowAmount plus flash loan fees.
+    function _invest(
+        bytes6 ilkId,
+        bytes6 seriesId,
+        uint128 amountToInvest,
+        uint128 borrowAmount,
+        uint128 minCollateral
+    ) internal returns (bytes12 vaultId) {
+        // TODO: Maybe check whether the series/ilkId is supported
```

Comment:
Yes, but in the external function. Parameters are checked as soon as possible, when we do it at all.

There are no data integrity checks in this contract because it has no storage, but for completeness if you would have data integrity checks those would go where data is modified instead.